src/util.ts

```diff
@@ -0,0 +1,106 @@
+import * as sourcegraph from 'sourcegraph'
+import { Metadata } from './link-preview-expander'
+import LRU from 'lru-cache'
+
+export function getWord(document: sourcegraph.TextDocument, range: sourcegraph.Range): string {
+    const lines = document.text?.split('\n')
+    let word = ''
+    if (lines && range) {
+        word = lines[range.start.line].slice(range.start.character, range.end.character)
+    }
+    return word
+}
+
+export function checkIsURL(maybeURL: string): boolean {
+    try {
+        const url = new URL(maybeURL)
+        return url.protocol === 'http:' || url.protocol === 'https:'
+    } catch {
+        return false
+    }
+}
+
+/**
+ * Strips trailing comma or period from URL
+ *
+ * @param maybeURL Possible URL string
+ */
+export function cleanURL(maybeURL: string): string {
+    const lastCharacter = maybeURL.slice(-1)
+    if (lastCharacter === '.' || lastCharacter === ',') {
+        return maybeURL.slice(0, -1)
+    }
+    return maybeURL
```

Comment:
Suggestion: simplification
```suggestion
    return maybeURL.replace(/[,\.]$/, '')
```